File: 0012-integer-to-roman/solution.py
class Solution:
    def intToRoman(self, num: int) -> str:
        result = ""
        while(num != 0):
            if num >= 1000:
                num -= 1000
                result += "M"

            if num >= 900 and num < 1000:
                num -= 900
                result += "CM"  

            if num >= 500 and num < 900:
                num -= 500
                result += "D"

            if num >= 400 and num < 500:
                num -= 400
                result += "CD"

            if num >= 100 and num < 400:
                num -= 100
                result += "C"
            if num >= 90 and num < 100:
                num -= 90
                result += "XC"

            if num >= 50 and num < 90:
                num -= 50
                result += "L"

            if num >= 40 and num < 50:
                num -= 40
                result += "XL"

            if num >= 10 and num < 40:
                num -= 10
                result += "X"

            if num == 9:
                num -= 9
                result += "IX"

            if num >= 5 and num < 9:
                num -= 5
                result += "V"

            if num == 4:
                num -= 4
                result += "IV"

            if num >= 1 and num < 4:
                num -= 1
                result += "I"
        return result

File: 0012-integer-to-roman/solution.cpp
class Solution {
public:
    string intToRoman(int num) 
    {
        string ans = "";
        
        while(num)
        {
            if(num >= 1000)
            {
                ans += "M";
                num -= 1000;
            }
            else if(num >= 900)
            {
                ans += "CM";
                num -= 900;
            }
            else if(num >= 500)
            {
                ans += "D";
                num -= 500;
            }
            else if(num >= 400)
            {
                ans +=  "CD";
                num -= 400;
            }
            else if(num >= 100)
            {
                ans += "C";
                num-= 100;
            }
            else if(num >= 90)
            {
                ans += "XC";
                num -= 90;
            }
            else if(num >= 50)
            {
                ans += "L";
                num -= 50;
            }
            else if(num >= 40)
            {
                ans += "XL";
                num-= 40;
            }
            else if(num >= 10)
            {
                ans += "X";
                num -= 10;
            }

            switch(num)
            {
                case 1: 
                    ans += "I";  num = 0; break;
                case 2:
                    ans += "II"; num = 0; break;
                case 3:
                    ans +=  "III"; num = 0; break;
                case 4:
                    ans +=  "IV"; num = 0; break;
                case 5:
                    ans +=  "V"; num = 0; break;
                case 6:
                    ans +=  "VI"; num = 0; break;
                case 7:
                    ans +=  "VII"; num = 0; break;
                case 8:
                    ans +=  "VIII"; num = 0; break;
                case 9:
                    ans +=  "IX"; num = 0; break;
            }

        }
        return ans;
    }
};
